controllers/chat/getMessageDetail.js

```diff
@@ -0,0 +1,31 @@
+const {StreamChat} = require('stream-chat');
+const {ResponseSuccess} = require('../../utils/Responses');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const getMessageDetail = async (req, res) => {
+  const {messageID} = req.params;
+
+  if (!messageID) {
+    return res.status(400).json({
+      message: 'Bad request'
+    });
+  }
+  const client = new StreamChat(process.env.API_KEY, process.env.SECRET);
+  let message_detail = await client.getMessage(messageID);
+  if (!message_detail) {
+    return res.status(400).json({
+      success: false,
+      message: `You do not have permission to access this content, or the content has been deleted`
+    });
+  }
+
+  return ResponseSuccess(res, 'Success', 200, {
+    message: message_detail
+  });
+};
+
+module.exports = getMessageDetail;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good overall, but there's a potential issue with error handling. The `client.getMessage(messageID)` function is likely to throw an exception if the message ID is invalid or if there's a network error. This isn't currently being handled, which could lead to unhandled promise rejections.

Consider wrapping the `client.getMessage(messageID)` call in a try-catch block to handle any exceptions that might be thrown.
```diff
  const client = new StreamChat(process.env.API_KEY, process.env.SECRET);
+ try {
  let message_detail = await client.getMessage(messageID);
+ } catch (error) {
+   return res.status(500).json({
+     success: false,
+     message: 'An error occurred while retrieving the message',
+     error: error.message
+   });
+ }
  if (!message_detail) {
    return res.status(400).json({
      success: false,
      message: `You do not have permission to access this content, or the content has been deleted`
    });
  }
```
This way, you can provide a more informative error message to the client and prevent the server from crashing due to unhandled exceptions.


<!-- This is an auto-generated comment by OSS CodeRabbit -->